src/server/comlib/ldcs_api_pipe.c

```diff
@@ -342,63 +349,60 @@ int ldcs_recv_msg_static_pipe(int fd, ldcs_message_t *msg, ldcs_read_block_t blo
      msg->data = NULL;
      return(rc);
   }
-  if (n < 0) _error("ERROR reading header from connection");
 
   if(msg->header.len>0) {
-    n = _ldcs_read_pipe(fdlist_pipe[fd].in_fd,msg->data,msg->header.len, LDCS_READ_BLOCK);
-    if (n == 0) 
-       return(rc);
-    if (n < 0) {
-       int error = errno;
-       err_printf("Error during read of pipe: %s (%d)\n", strerror(error), error);
+    n = _ldcs_read_pipe(fdlist_pipe[fd].in_fd,msg->data,msg->header.len, LDCS_READ_BLOCK, &err);
+    if (err) {
+       err_printf("Error during read of pipe: %s (%d)\n", strerror(err), err);
        return -1;
     }
+    if (n == 0)
+       return(rc);
     if (n != msg->header.len) {
-       int error = errno;
        err_printf("Partial read on pipe.  Got %u / %u: %s (%d)\n",
                   (unsigned) n, (unsigned) msg->header.len,
-                  strerror(error), error);
+                  strerror(err), err);
        return -1;
     }
 
   } else {
     *msg->data = '\0';
   }
 
-  debug_printf3("received message of type: %s len=%d data=%s ...\n",
+  debug_printf3("received message of type: %s len=%zu data=%s ...\n",
 	       _message_type_to_str(msg->header.type),
 	       msg->header.len, msg->data );
 
   return(rc);
 }
 
-int _ldcs_read_pipe(int fd, void *data, int bytes, ldcs_read_block_t block ) {
+size_t _ldcs_read_pipe(int fd, void *data, size_t bytes, ldcs_read_block_t block, int *err ) {
 
-  int         left,bsumread;
-  ssize_t     btoread, bread;
-  char       *dataptr;
+  ssize_t    rc;
+  size_t     btoread, bread,left=bytes,bsumread=0;
+  char       *dataptr= (char*) data;
   int print_count = 512;
-  left      = bytes;
-  bsumread  = 0;
-  dataptr   = (char*) data;
 
   while (left > 0)  {
     btoread    = left;
     debug_printf3("before read from fifo %d, bytes_to_read = %ld\n", fd, btoread);
-    bread      = read(fd, dataptr, btoread);
-    if(bread<0) {
+    errno=0;
+    rc = read(fd, dataptr, btoread);
+    *err = errno;
```

Comment:
As an example from the above review comment, here we set *err even if it's not a hard error.